### PR TITLE
Migrate Travis WCT rules into Makefile

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Working towards https://github.com/web-platform-tests/wpt.fyi/issues/163

Consolidates the WCT and webdriver test behaviours as `OUTSIDE_DOCKER` more explicitly.
`go_webdriver_test` now defaults to the `node_module` binaries as provided by `web-component-tester` install, making the `install.sh` script redundant.

## Review Information
Check Travis runs for `integration_test`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -93,11 +68,11 @@ script:
     fi
   - | # Run tests
     if [ "${MAKE_TEST_TARGET}" != "" ]; then
-      docker exec -t -u $(id -u $USER):$(id -g $USER) "${DOCKER_INSTANCE}" make "${MAKE_TEST_TARGET}"
-    fi
-  - | # Run WCT tests
-    if [ "${WCT}" == "1" ]; then
-      cd webapp && npm test
+      if [ ${OUTSIDE_DOCKER} == true ]; then
```
Comment (mdittmer) [inline .travis.yml]: nit: quote lhs & rhs

File: Makefile [hunk 2]
```diff
@@ -12,21 +12,21 @@
...
 
 # WPTD_PATH will have a trailing slash, e.g. /home/user/wpt.fyi/
 WPTD_PATH := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 WPTD_GO_PATH ?= $(GOPATH)/src/github.com/web-platform-tests/wpt.fyi
-WEBDRIVER_PATH ?= $(WPTD_GO_PATH)/webdriver
-BROWSERS_PATH ?= $(HOME)/browsers
-SELENIUM_PATH ?= $(BROWSERS_PATH)/selenium
-FIREFOX_PATH ?= $(BROWSERS_PATH)/firefox/firefox
-GECKODRIVER_PATH ?= $(BROWSERS_PATH)/geckodriver
+NODE_SELENIUM_PATH=$(WPTD_PATH)webapp/node_modules/selenium-standalone/.selenium/
```
Comment (mdittmer) [inline Makefile]: nit: Is there a reaons we use `FOO=bar` but `FOO ?= bar`? @Hexcles and @arobins do you know any make details for that should tend towards one or the other consistent style?

File: Makefile [hunk 3]
```diff
@@ -37,96 +37,114 @@ prepush: build test lint
...
 
-go_lint: go_deps go_test_tag_lint
+go_lint: go_deps golint_deps go_test_tag_lint
 	@echo "# Linting the go packages..."
-	@cd $(WPTD_GO_PATH); golint -set_exit_status api/
-	@cd $(WPTD_GO_PATH); golint -set_exit_status revisions/
-	@cd $(WPTD_GO_PATH); golint -set_exit_status shared/
-	@cd $(WPTD_GO_PATH); golint -set_exit_status util/
-	@cd $(WPTD_GO_PATH); golint -set_exit_status webapp/
+	cd $(WPTD_GO_PATH); golint -set_exit_status api/
```
Comment (mdittmer) [inline Makefile]: Just out of curiosity. What was the `@` being dropped here doing before?

File: Makefile [hunk 4]
```diff
@@ -138,11 +156,26 @@ webapp_deploy_staging: bower_components env-BRANCH_NAME
...
+		sudo apt-get install --assume-yes --no-install-suggests xvfb; \
+		export DISPLAY=99; Xvfb :99 -screen 0 1024x768x24 -ac +extension GLX +render -noreset & \
+	fi
+
+gcloud-%: gcloud
+	gcloud components install --quiet $*
+
+node-%: node
+	@ echo "# Installing $*..."
+	cd $(WPTD_PATH)webapp; node -p "require('$*/package.json').version" 2>/dev/null || npm install --no-save $*
```
Comment (mdittmer) [inline Makefile]: Maybe add a comment explaining `node -p .... 2>/dev/null`. Sweet hack, though :smile:

File: Makefile [hunk 5]
```diff
@@ -37,96 +37,116 @@ prepush: build test lint
...
+
+node: curl
+	if [[ "$$(which node)" == "" ]]; then \
+		curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -; \
+		sudo apt-get install -y nodejs; \
+	fi
+
+npm: apt-get-npm
+
+gcloud: python curl
```
Comment (mdittmer) [inline Makefile]: Need dep on `gpg` here.

Comment (mdittmer) [reply Makefile]: Done.

File: .travis.yml (hunk 1 above)
Comment (lukebjerring) [reply .travis.yml]: Removed.

**Pull request merged**